Core.ApplicationServices/GDPR/DataProcessingRegistrationApplicationService.cs

```diff
@@ -291,19 +294,54 @@ public Result<(DataProcessingRegistration registration, IEnumerable<DataProcessi
         {
             return Modify<DataProcessingRegistration>(id, registration =>
             {
-                registration.SetHasSubDataProcessors(state);
+                var result = registration.SetHasSubDataProcessors(state);
+                var removedSubDataProcessors = result.RemovedSubDataProcessors.ToList();
+                if (removedSubDataProcessors.Any())
+                {
+                    _sdpRepository.RemoveRange(removedSubDataProcessors);
```

Comment:
is this check necessary since repository also checks if any entities are present?

Reply:
that's true.. will remove the condition